src/editions/transitioning.md

```diff
@@ -1,11 +1,100 @@
 # Transitioning your code to a new edition
 
-In general, to update your code to a new edition, the process
-will look like this:
+Transitioning between editions is built around lints. Fundamentally, the
+process works like this:
 
-* Make sure your build is warning-free.
-* Update or add the `edition` key in `Cargo.toml`
-* Fix any warnings, possibly by running `rustfix`.
+* Get your code compiling with no warnings.
+* Opt in to the new edition.
+* Fix any new warnings that may result.
 
-At this time, the `edition` key does not yet work, and `rustfix` is in a
-larval state. This section will update as our story gets better here.
\ No newline at end of file
+Luckily, we've been working on a tool to help assist with this process,
+`rustfix`. It can take suggestions from the compiler and automatically
+re-write your code to comply with new features and idioms.
+
+> `rustfix` is still quite young, and very much a work in development. But it works
+> for the basics! We're working hard on making it better and more robust, but
+> please bear with us for now.
+
+## Installing rustfix
+
+You can get `rustfix` from crates.io. Given that you're probably using Cargo,
+you'll want to run this:
+
+```shell
+$ cargo install cargo-fix --git https://github.com/rust-lang-nursery/rustfix
+```
+
+And that's it!
+
+## Prepare for the next edition
+
+Before we talk about how to move to the new edition, a reminder:
+
+
+* New editions may have features that are only available in the new edition.
+  For example, `async`/`await` is available in Rust 2018, but not Rust 2015.
+* It's our intention that the migration to new editions is as smooth an experience
+  as possible. It's considered a bug if it's difficult to upgrade your crate to
```

Comment:
> difficult

That is a super opaque und undefined metric. It's cool if you don't want to be more specific, but I'd add something like

> What do we mean by "difficult"? Migration code in macros for example: The compiler often can't directly determine how the code might be used and thus can't provide a way to upgrade in a straightforward way.

Reply:
I personally like this sort of being opaque and undefined in the sense that if you *feel* it's difficult that's a bug we want to know about and fix. It'll definitely be a lot of work for us to bottom out "what makes it difficult" in some situations, but no one should ever get the impression that an edition upgrade is difficult ideally

Reply:
Yes, I think for now I'd like to leave this vague. As we get closer to shipping, we'll know better what those pain points are.